### DISCUSSION TITLE
ISR (Incremental Static Regeneration) is not working in dynamic routing pages

### DISCUSSION THREAD

**Discussion by summerisgood1** [vercel/next.js#41711] [Help] [answered]
Demo site: https://oscaryiu-com-i1dy-git-feat-view-count-15077693d.vercel.app/
PR: https://github.com/15077693d/oscaryiu.com/pull/3

In page /blog and /, Incremental Static Regeneration is perfectly working.
1. Go to https://oscaryiu-com-i1dy-git-feat-view-count-15077693d.vercel.app/blog/motivation-on-oscaryiu-2-0
2. Go to https://oscaryiu-com-i1dy-git-feat-view-count-15077693d.vercel.app/ 
3. View count change after refresh

However, In page /blog/[...slug] and /tags/[tag] is not working
1. Go to https://oscaryiu-com-i1dy-git-feat-view-count-15077693d.vercel.app/blog/motivation-on-oscaryiu-2-0
2. Refresh the page
3. View count doesn't change after refresh

According to https://stackoverflow.com/questions/67787456/what-is-the-difference-between-fallback-false-vs-true-vs-blocking-of-getstaticpa, I update getStaticPaths to return {fallback: true}.

However, it still not working. 
May I get some help?

**Comment by icyJoseph** [comment, marked as answer]
A bit of a shot in the dark here, but you have a lot of async and IO operations, in the `getStaticProps` function for `/pages/blog/[...slug]` - Would it be possible that ISR is failing to run the function once deployed on the Vercel servers? When that happens rather than showing a 404 or some error page, Next.js keeps the last working visual of the page.

Try to add some error handling and logging to figure out if `getStaticProps` is returning or if one of the steps is throwing. Again, just guessing here, but these lines look suspicious:

```js
  // rss
  if (allPosts.length > 0) {
    const rss = generateRss(allPosts)
    fs.writeFileSync('./public/feed.xml', rss)
  }

```
You might want to use `process.cwd()` and double check if you can really write to files like that in Vercel. https://nextjs.org/docs/api-reference/data-fetching/get-static-props#reading-files-use-processcwd

**Reply by summerisgood1** [reply]
OMG....
![Screenshot 2022-10-24 at 5 59 35 PM](https://user-images.githubusercontent.com/51513183/197501193-d23fab21-6dfa-4cbe-8ed5-96072f5208ad.png)

**Reply by summerisgood1** [reply]
I add a try catch statement on generateRss code block and it's working now.
But I'm confused that why the path on build time and ISR rebuild time are different. Thanks for your help!

**Reply by summerisgood1** [reply]
<img width="756" alt="Screenshot 2022-10-24 at 7 22 20 PM" src="https://user-images.githubusercontent.com/51513183/197514913-ef310e84-1197-4726-aeed-dc97dac4d5e4.png">
<img width="507" alt="Screenshot 2022-10-24 at 7 23 04 PM" src="https://user-images.githubusercontent.com/51513183/197514950-ac98dcfa-beb2-4e37-9734-8de200e31b62.png">

**Reply by icyJoseph** [reply]
Roughly speaking _**Next.js in Vercel**_, tries to keep resources consumption to a minimum, and that's why it runs the `getStaticProps`, and all of the API routes + `GSSP`, calls as serverless tasks, from which it then collects their results. So you shouldn't expect API Routes and other data fetching functions to run on the same context as the server, because the runtime might be using Next.js in a whole different way than you do locally.
